### DISCUSSION TITLE
Licence valid from a set start date

### DISCUSSION THREAD

**Discussion by willnSSL** [keygen-sh/community#148] [Q&A] [answered]
Is it possible to set a licence to be valid from a particular date? Instead of being valid starting on creation, or allowing the user to set the start date, via 'on first validation', is it possible to define, when we create or update the licence, a date in the future that the licence will be valid from?

Thanks!

**Comment by ezekg** [comment]
Yes. You can set a `FROM_FIRST_VALIDATION` expiration basis: https://keygen.sh/docs/api/policies/#policies-object-attrs-expirationBasis. Or to set an arbitrary future date, just use addition to add your time delta to the license expiry during creation:

```js
{
  "data": {
    "type": "license",
    "attributes": {
      "expiry": "my-custom-date"
    }
  }
}
```

Or just wait and create the license just-in-time.

**Reply by willnSSL** [reply]
I'm not sure I understand correctly - With `FROM_FIRST_VALIDATION`, is the user not free to start the licence whenever they please? Is there any way to prevent validation until the start date?

**Reply by ezekg** [reply]
No. Can you explain your use case in more detail? I've never heard of a use case before where you distribute a license key to a user but they aren't allowed to use it until some arbitrary date in the future.

**Reply by willnSSL** [reply]
Sometimes customers want to enable particular licenced features on a short-term basis for particular events in advance and they want to avoid accidentally starting the timer on these licences ahead of time. With specified start dates they can just load all the licences for a year onto the machines in question, and the capabilities will expand as needed for their planned events.

Having short-term licences that they are responsible for managing the start times of is probably the next best thing though!

**Reply by ezekg** [reply, marked as answer]
Since this is such a niche use-case, I'd suggest adding a `notValidUntil` key into the license's metadata and validating the start date at the application level by looking at `data.attributes.metadata.notValidUntil`.

**Reply by willnSSL** [reply]
Oh that's a good idea, thanks!
